### PR TITLE
ISLE Sprint Work

### PR DESCRIPTION
**Pull request by dannylamb**
Supersedes https://github.com/Islandora-Devops/isle-dc/pull/126 and https://github.com/Islandora-Devops/isle-dc/pull/123

This represents several tasks from the latest ISLE sprint and is ready for testing.
- Secrets
- Using a single variable for Domain
- Only one alias, which we've found is only neccessary for when you don't own a domain
- Better Drupal config for https and reverse proxy

This requires https://github.com/Islandora-Devops/isle-buildkit/pull/87 to test

### Testing Instructions

- Pull down and build https://github.com/Islandora-Devops/isle-buildkit/pull/87
- `cp sample.env .env` if you haven't already
- Set the following `.env`
```
USE_SECRETS=true
DOMAIN=islandora.traefik.me
REPOSITORY=local
```
- `make`
- `docker-compose up -d`

Everything should come up you can smoke test uploading content.  I've tested fcrepo indexing, derivatives, the openseadragon viewer, and matomo.  Everything seems to check out.

### CODE REVIEW

**Comment by nigelgbanks** [comment]
I might be misunderstanding the JWT usage, but I thought that all the public keys used by micros-services must be paired against the the JWT private key used by the Drupal JWT module. 

If that is the case aren't all the public keys used by the micro-services forced to use the same public key file value? If this is the case couldn't we change all the secrets to mount the one file instead of duplicating it? It would reduce the maintenance burden for those replacing the files locally.

Not sure if this applies to the token value as well?

**Comment by nigelgbanks** [comment]
Might just be me, but I'm digging into

```bash
drupal_1      | ERROR 1045 (28000): Access denied for user 'root'@'isle-dc_drupal_1.isle-dc_default' (using password: YES)
```

**Comment by dannylamb** [comment]
That happens to me if the secrets aren't getting decoded.  Your password is probably `secret:/run/secrets/MYSQL_ROOT_PASSWORD` and not it's actual default value of `password`. We recently merged in changes to isle-buildkit. Maybe you just need to `docker-compose pull`?  It happened to both me and @g7morris when working on it.  At one point I got stuck with cached images and had to `docker system prune -a` to get it going.  Hopefully it doesn't come to that.  You can confirm you've got the latest by looking for the `00-load-secrets` script in `/etc/cont-init.d`.

**Comment by nigelgbanks** [comment]
> Might just be me, but I'm digging into
> 
> ```shell
> drupal_1      | ERROR 1045 (28000): Access denied for user 'root'@'isle-dc_drupal_1.isle-dc_default' (using password: YES)
> ```

Was just me 
download

File: docker-compose.sprint4.yml [hunk 1]
```diff
@@ -0,0 +1,445 @@
+version: '3.7'
```
Comment (nigelgbanks) [inline docker-compose.sprint4.yml]: Should this be checked in?

**Comment by dannylamb** [comment]
And yeah, I'll switch it up to mount the same secret.  At least that'll cut out some of the tedium until we figure out how to get both HTTPS and JWT handled once at the proxy and that's it.

File: docker-compose.sprint4.yml (hunk 1 above)
Comment (dannylamb) [reply docker-compose.sprint4.yml]: Nope.  Not at all :sweat_smile:

**Comment by nigelgbanks** [comment]
Not sure if related cause I switched to the docker hub images to continue testing, but I seem to have lost the ability to extract text from uploaded PDF(s). Hypercube is getting the messages though.

```bash
hypercube_1   | 172.30.0.15 - - [20/Nov/2020:14:57:15 +0000] "GET / HTTP/1.1" 200 25117 "-" "Apache-HttpClient/4.5.3 (Java/1.8.0_252)" "-"
hypercube_1   | [2020-11-20 15:00:09] app.INFO: Matched route "{route}". {"route":"GET_","route_parameters":{"_controller":"hypercube.controller:get","_route":"GET_"},"request_uri":"http://hypercube:8000/","method":"GET"} []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: Checking for guard authentication credentials. {"firewall_key":"default","authenticators":1} []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: Checking support on guard authenticator. {"firewall_key":"default","authenticator":"Islandora\\Crayfish\\Commons\\Syn\\JwtAuthenticator"} []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: Calling getCredentials() on guard authenticator. {"firewall_key":"default","authenticator":"Islandora\\Crayfish\\Commons\\Syn\\JwtAuthenticator"} []
hypercube_1   | [2020-11-20 15:00:09] crayfish.syn.jwt_authentication.DEBUG: Token: eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpYXQiOjE2MDU4ODQ0MDksImV4cCI6MTYwNTg5MTYwOSwid2ViaWQiOiIxIiwiaXNzIjoiaHR0cHM6XC9cL2lzbGFuZG9yYS50cmFlZmlrLm1lIiwic3ViIjoiYWRtaW4iLCJyb2xlcyI6WyJhdXRoZW50aWNhdGVkIiwiYWRtaW5pc3RyYXRvciIsImZlZG9yYWFkbWluIl19.SRuYvKcsn3MYcY5Oa2dq-7GoigX4U6TAfRFlEFVAYktf9RDk3-ODdLATRfPZVyORHUIU4hy5Bxszy9GjNcCHy8RVGR6ovayIitMNQOFWGaWnYxx_36geFQ7Ksec1rza-GyzhRA2fWr6-op5rFU5dJu7LmNp9_oOEfBPKd96xbG1h6OJ9QT9XpJzXiEiv1WVtrRGk2H4hPqGW5Nc495joN7S10jSnGE8N_3YCP5A6cZ0bLfmP_ZDYoENcchWtf6xDPRjvXfcw6cVzjDSZwV7FIJjNTit-69imGH6CLguupHnDwew_n2G4iLaAy32IuyLOanByEUeoT9yC2O7OCnr9XA [] []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: Passing guard token information to the GuardAuthenticationProvider {"firewall_key":"default","authenticator":"Islandora\\Crayfish\\Commons\\Syn\\JwtAuthenticator"} []
hypercube_1   | [2020-11-20 15:00:09] app.INFO: Guard authentication successful! {"token":"[object] (Symfony\\Component\\Security\\Guard\\Token\\PostAuthenticationGuardToken: PostAuthenticationGuardToken(user=\"admin\", authenticated=true, roles=\"authenticated, administrator, fedoraadmin\"))","authenticator":"Islandora\\Crayfish\\Commons\\Syn\\JwtAuthenticator"} []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: Guard authenticator set no success response: request continues. {"authenticator":"Islandora\\Crayfish\\Commons\\Syn\\JwtAuthenticator"} []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: Remember me skipped: it is not configured for the firewall. {"authenticator":"Islandora\\Crayfish\\Commons\\Syn\\JwtAuthenticator"} []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: > GET / [] []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: Got Content-Type: {"type":"application/pdf"} []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: Executing command: {"cmd":"pdftotext  - -"} []
hypercube_1   | [2020-11-20 15:00:09] app.DEBUG: < 200 [] []
hypercube_1   | 172.30.0.15 - - [20/Nov/2020:15:00:09 +0000] "GET / HTTP/1.1" 200 28 "-" "Apache-HttpClient/4.5.3 (Java/1.8.0_252)" "-"

```

**Comment by nigelgbanks** [comment]
Actually that seem to have affects all derivatives

**Comment by dannylamb** [comment]
Probably related if the reverse_proxy_ips aren't set.  Then Drupal won't put `https` into the urls it published in messages and Alpaca complains because it's getting redirect responses when requesting `http` versions of things.

**Comment by nigelgbanks** [comment]
> Probably related if the reverse_proxy_ips aren't set. Then Drupal won't put `https` into the urls it published in messages and Alpaca complains because it's getting redirect responses when requesting `http` versions of things.

Cheers!

Makes sense since the switch to the Docker hub images, I'll manually modify my `settings.php` to continue testing.

**Comment by nigelgbanks** [comment]
That concludes it, it's working well :+1: I'd be ready to merge this after the removal of that `sprint4` yml file and the amalgamation of the public keys.

**Comment by dannylamb** [comment]
@nigelgbanks Got all that sorted out.  Tested it and things seem fine with sharing the same secret file for the JWT keys and token.

**Review by nigelgbanks — APPROVED** [review]

**Pull request merged**